src/ray/gcs/tables.cc

```diff
@@ -746,6 +746,81 @@ Status TaskLeaseTable::Subscribe(const JobID &job_id, const ClientID &client_id,
   return Table<TaskID, TaskLeaseData>::Subscribe(job_id, client_id, on_subscribe, done);
 }
 
+std::vector<ActorID> SyncGetAllActorID(redisContext *redis_context,
+                                       const std::string &table_prefix) {
+  std::vector<ActorID> actor_id_list;
+  size_t cursor = 0;
+  do {
+    auto r = redisCommand(redis_context, "SCAN %d match %s* count 100", cursor,
+                          table_prefix.c_str());
+    auto reply = reinterpret_cast<redisReply *>(r);
+    RAY_CHECK(reply != nullptr && reply->type == REDIS_REPLY_ARRAY);
+    RAY_CHECK(reply->elements == 2);
+
+    // current cursor
+    redisReply *cursor_reply = reply->element[0];
+    RAY_CHECK(cursor_reply != nullptr && cursor_reply->type == REDIS_REPLY_STRING);
+    cursor = std::stoi(std::string(cursor_reply->str, cursor_reply->len));
+
+    // actor ids
+    redisReply *array_reply = reply->element[1];
+    RAY_CHECK(array_reply != nullptr && array_reply->type == REDIS_REPLY_ARRAY);
+    for (size_t i = 0; i < array_reply->elements; ++i) {
+      redisReply *id_reply = array_reply->element[i];
+      RAY_CHECK(id_reply != nullptr && id_reply->type == REDIS_REPLY_STRING);
+      auto id_with_prefix = std::string(id_reply->str, id_reply->len);
+      // The key of actor_checkpoint table and actor_checkpoint_id table have the same
+      // prefix of `ACTOR`, so we should check the length of the key to filter them.
+      if (id_with_prefix.size() == table_prefix.size() + ActorID::Size()) {
+        auto id = ActorID::FromBinary(id_with_prefix.substr(table_prefix.size()));
+        actor_id_list.emplace_back(id);
```

Comment:
Remove duplicate data. As the [document](https://redis.io/commands/scan) described:

However because SCAN has very little state associated (just the cursor) it has the following drawbacks:
A given element may be returned multiple times. It is up to the application to handle the case of duplicated elements, for example only using the returned elements in order to perform operations that are safe when re-applied multiple times.